src/main/java/com/mongodb/kafka/connect/source/heartbeat/HeartbeatManager.java

```diff
@@ -93,4 +113,75 @@ public HeartbeatManager(
     }
     return Optional.empty();
   }
+
+  @Override
+  public void close() throws Exception {
+    if (!isClosed.getAndSet(true) && heartbeatConsumer != null) {
+      heartbeatConsumer.shutdown();
+    }
+  }
+
+  public class HeartbeatConsumer implements Runnable {
+    private final AtomicBoolean running;
+    private final KafkaConsumer<Bytes, Bytes> consumer;
+
+    public HeartbeatConsumer(final List<String> bootStrapServers) {
+      this.consumer = tryCreateConsumer(String.join(",", bootStrapServers)).orElse(null);
+      this.running = new AtomicBoolean(false);
+      if (consumer != null) {
+        new Thread(this).start();
+        LOGGER.info("Start heartbeat offset consumer");
+      }
+    }
+
+    public void run() {
```

Comment:
This is a runnable class and is called in the constructor via Thread.start: 

```
if (consumer != null) {
    new Thread(this).start();
    LOGGER.info("Start heartbeat offset consumer");
}
```

Reply:
Thanks! I completely missed this. Since the class was moved, I replied in a [different place](https://github.com/mongodb/mongo-kafka/pull/58#discussion_r592809894).